packages/devtools-proxy-support/src/system-ca.ts

```diff
@@ -50,6 +51,54 @@ export function mergeCA(...args: (NodeJSCAOption | undefined)[]): string {
   return [...ca].join('\n');
 }
 
+const pemWithParsedCache = new WeakMap<
+  string[],
+  { pem: string; parsed: X509Certificate | null }[]
+>();
+// TODO(COMPASS-8253): Remove this in favor of OpenSSL's X509_V_FLAG_PARTIAL_CHAIN
+// See linked tickets for details on why we need this (tl;dr: the system certificate
+// store may contain intermediate certficiates without the corresponding trusted root,
+// and OpenSSL does not seem to accept that)
+export function removeCertificatesWithoutIssuer(ca: string[]): {
+  ca: string[];
+  messages: string[];
+} {
+  const messages: string[] = [];
+  let caWithParsedCerts =
+    pemWithParsedCache.get(ca) ??
+    ca.map((pem) => {
+      let parsed: X509Certificate | null = null;
+      try {
+        parsed = new X509Certificate(pem);
+      } catch (err: unknown) {
+        messages.push(
+          `Unable to parse certificate: ${
+            err && typeof err === 'object' && 'message' in err
+              ? String(err.message)
+              : String(err)
+          }`
```

Comment:
No strong opinion here and mostly out of curiosity, can this be rewritten more concisely as: `Unable to parse certificate: ${ err?.message || err }`? I guess it will require replacing the type for `err` with `any`, but other than that are there downsides?

Reply:
> I guess it will require replacing the type for `err` with `any`, but other than that are there downsides?

Nah, you pretty much got it – it's not more than a general tendency to avoid `any` whenever we can 🙂